### ISSUE TITLE
Back/Forward navigation don't work as expected for iframe navigations

### ISSUE THREAD

**Issue by Desz5** [MicrosoftEdge/WebView2Feedback#1492]
Hello,
I'm using VB.NET and Winforms and i'm trying to go forward in WebView2, i can go back but it doesn't allow me to navigate forward.
The page that i'm navigating has frames.
I've put the following code in the HistoryChanged event, the SourceChanged event and in the FrameNavigationCompleted event.

```
If WebView_Browser1.CanGoForward = True Then
MsgBox("yes")
Else
MsgBox("no")
End If
```
The event always returns 'no' even if i can goforward using ALT+Right.
and when i call WebView_Browser1.goForward() it never works.
Everything is working great with goBack.
With Edge i can navigate forward and back.
Am i doing something wrong or is this a bug?
Thank you!

[AB#34509102](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/34509102)

**Comment by champnic** [comment]
Hey @Desz5 - I just tried this in a sample app and it works fine for me. It's possible that you can't navigate forwards if it's just a frame navigation. Are you getting a "NavigationCompleted" event? Does this work if you navigate to top-level sites?

**Comment by Desz5** [comment]
Hey @champnic - top-level navigation works. But when i'm testing on a page that has a frame i can't go forward.
I'm getting a "FrameNavigationCompleted" event, not a "NavigationCompleted" event. It will navigate forward using the old webbrowser. Just not with the WebView2 browser. Thank you!

**Comment by champnic** [comment]
@Desz5 Thanks for confirming! If you load that page in Edge or Chrome do you get a similar behavior, or does it let you do a forward navigation just for the frame?

It's possible that WebBrowser (based on the Internet Explorer engine) and WebView2 (based on Chromium) have different notions about going forwards and backwards.

**Comment by Desz5** [comment]
@champnic if i load that page in Edge or Chrome i can do forward navigation just for the frame. Thank you!

**Comment by champnic** [comment]
Do you have a sample page or website that exhibits this behavior that I can try out?

**Comment by Desz5** [comment]
It's called Exact, you need to login to get access. We use it at work. It's a website that uses a top frame and a bottom frame and the top frame remains steady, and the bottom frame navigates. Here's a screenshot if that helps: https://www.exact.com/images/Releases/2019/spring/afronden-naar-standaard-seriegrootte.jpg Thank you!

**Comment by champnic** [comment]
@Desz5 I was able to reproduce the issue. I'm going to change this to a bug and open it on our backlog. Thanks!

**Comment by Desz5** [comment]
Thank you!

**Comment by champnic** [comment]
This is fixed in runtimes 100.0.1163.0+. Thanks!
